lib/plug/adapters/cowboy2/conn.ex

```diff
@@ -0,0 +1,109 @@
+defmodule Plug.Adapters.Cowboy2.Conn do
+  @behaviour Plug.Conn.Adapter
+  @moduledoc false
+
+  def conn(req) do
+    path = :cowboy_req.path(req)
+    host = :cowboy_req.host(req)
+    port = :cowboy_req.port(req)
+    meth = :cowboy_req.method(req)
+    hdrs = :cowboy_req.headers(req)
+    qs = :cowboy_req.qs(req)
+    peer = :cowboy_req.peer(req)
+    {remote_ip, _} = peer
+
+    req = Map.put(req, :plug_read_body, false)
+
+    %Plug.Conn{
+      adapter: {__MODULE__, req},
+      host: host,
+      method: meth,
+      owner: self(),
+      path_info: split_path(path),
+      peer: peer,
+      port: port,
+      remote_ip: remote_ip,
+      query_string: qs,
+      req_headers: to_headers_list(hdrs),
+      request_path: path,
+      scheme: String.to_atom(:cowboy_req.scheme(req))
+    }
+  end
+
+  def send_resp(req, status, headers, body) do
+    headers = to_headers_map(headers)
+    status = Integer.to_string(status) <> " " <> Plug.Conn.Status.reason_phrase(status)
+    req = :cowboy_req.reply(status, headers, body, req)
+    {:ok, nil, req}
+  end
+
+  def send_file(req, status, headers, path, offset, length) do
+    %File.Stat{type: :regular, size: size} = File.stat!(path)
```

Comment:
Shouldn't we read the file only if length is `:all`? Maybe even something like:

```
def send_file(req, status, headers, path, offset, :all) do
  %File.Stat{type: :regular, size: size} = File.stat!(path)
  send_file(req, status, headers, path, offset, size)
end
```

Although I guess we are using File.stat! as a check if the file exists...

Reply:
This is how it works with the Cowboy adapter, so I'm going to leave it for now.